### DISCUSSION TITLE
multiple destination MSG

### DISCUSSION THREAD

**Discussion by sameh-farouk** [threefoldtech/rmb-rs#2] [General]
> dst: Twin ID to handle the request (can be multiple)

> if one or more twins are invalid, the message can't be processed, and an error message is pushed directly to the $ret queue. with details of the error.

We need to be clear from the beginning on how the multiple messages will be decomposed, and handled separately, also on how RMB would retry only failing one.
The go version was handled this **inappropriately** [see here](https://github.com/threefoldtech/rmb_go/issues/38), I fixed that in the V version but am willing to see a section explaining the flow for this feature.

Also, we can discuss if this feature can be **dropped** to simplify the logic of RMB and make it focus on doing only **a few things just well** as I didn't see yet useful use cases for the multiple destination msg in production (we could discuss what this feature bring to the RMB too) and the client would just create multiple dest message as multiple messages, anyway, it expects **multiple responses**.

**Comment by muhamadazmy** [comment]
Yes, i totally forgot to update the document about this. I will update the doc but my idea was as this

- msg is stored as (SET backlog.$id <msg>) in redis
- for each dst twin you push to the forward queue a `$id:$dst` string. 
- Workers recieve this, they then can retrieve the message body from redis, and then try to send to this specific destinations 
- this will allow a message to be send (and retired) to multiple destinations at the same time

**Reply by sameh-farouk** [reply]
Still dropping this Would be a valid option unless there are some reasons (needs) for why we would support that. it won't save the bandwidth (client and RMB would be on same machine), or improve the performance? so we need to Discuss (why) we would implement this, if the client can simply send message to multiple twins, by pushing the same messages with different dest without the added logic/hassle
